File: app/src/main/java/de/mchllngr/androidboilerplatekotlin/injection/ApplicationComponent.java
package de.mchllngr.androidboilerplatekotlin.injection;

import javax.inject.Singleton;

import dagger.Component;
import de.mchllngr.androidboilerplatekotlin.module.main.MainPresenter;

/**
 * Dagger2-component for the whole application.
 */
@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void inject(MainPresenter presenter);
}

File: app/src/main/java/de/mchllngr/androidboilerplatekotlin/injection/ApplicationComponent.kt
package de.mchllngr.androidboilerplatekotlin.injection

import dagger.Component
import de.mchllngr.androidboilerplatekotlin.module.main.MainPresenter
import javax.inject.Singleton

/**
 * Dagger2-component for the whole application.
 */
@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    fun inject(presenter: MainPresenter)
}
